### PR TITLE
Allow project selection in projman and integrate into dashboard

### PR DESCRIPTION
**Pull request by jrbotros**
This makes the project management view accessible in the dashboard dropdown links for authorized users, and allows them to navigate to any project within the interface.

[screenshots to come]

### CODE REVIEW

File: orchestra/static/orchestra/common/js/orchestra.routes.js [hunk 1]
```diff
@@ -30,7 +30,7 @@
         controllerAs: 'vm',
         templateUrl: $static('/static/orchestra/task/partials/task.html')
       })
-      .when('/project/:projectId', {
+      .when('/project/:projectId?', {
```
Comment (marcua) [inline orchestra/static/orchestra/common/js/orchestra.routes.js:33]: why the `?`?

File: orchestra/static/orchestra/project_management/js/data_service.js [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
+      getAllProjects: function() {
+        var dataService = this;
+        dataService.dataReady = orchestraApi.allProjects().then(function(response) {
+          response.data.forEach(function(project) {
+            dataService.allProjects[project.id] = project;
+          });
+        });
+        return dataService.dataReady;
+      },
+      changeProject: function(projectId) {
```
Comment (marcua) [inline orchestra/static/orchestra/project_management/js/data_service.js:37]: any chance you somehow have a request out to modify a project via projman, and then change the project, which results in unexpected behavior?

File: orchestra/interface_api/project_management/views.py [hunk 3]
```diff
@@ -29,6 +32,17 @@ def is_project_admin(user):
     return user.groups.filter(name='project_admins').exists()
 
 
+class IsProjectAdmin(permissions.BasePermission):
+    def has_permission(self, request, view):
+        return is_project_admin(request.user)
+
+
+class ProjectList(generics.ListCreateAPIView):
```
Comment (marcua) [inline orchestra/interface_api/project_management/views.py:31]: any way to paginate this?  presumably this list gets looooong :)

File: orchestra/templates/orchestra/base.html [hunk 4]
```diff
@@ -92,6 +89,13 @@
               </li>
 
               <li>
+                <a href="/orchestra/app/project/">
```
Comment (marcua) [inline orchestra/templates/orchestra/base.html:93]: only if they are a project admin

Comment (jrbotros) [inline orchestra/templates/orchestra/base.html:93]: ✅

File: orchestra/static/orchestra/common/js/orchestra.routes.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/common/js/orchestra.routes.js:33]: If you go to the URL from the dashboard there's no project ID attached, so you get a default "select a project" screen

File: orchestra/interface_api/project_management/views.py (hunk 3 above)
Comment (jrbotros) [inline orchestra/interface_api/project_management/views.py:31]: I don't know how I'd do that on the UX side if they're in a dropdown, but open to suggestions!

File: orchestra/static/orchestra/project_management/js/data_service.js (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/project_management/js/data_service.js:37]: The select is now disabled when data is being updated, and as an added bonus `changeProject` is wrapped in the `then` clause of the update promise!

**Pull request merged**